【課題提出】其の四 『WEBページの基本的なレイアウト2』再　安藤

andoazumi opened the pull request:
@tommyKazuhito 
@lineatakano 

課題其の四 『WEBページの基本的なレイアウト2』の再チャレンジを「マージンは上・左、culmnで作るものはflexを使う縛り」で行いました。お手隙の際にご確認の程よろしくお願い申し上げます。
制作時間3時間

今回は特に変に冒険せず二回目だったので、すんなり作成できました。
二番目以降にマージンをつけるための記述は自分的にどの方法がいいのか検討するために、次の課題等で別の書き方で試そうと考えています。

andoazumi committed:
[Add]其の四 『WEBページの基本的なレイアウト2』の作成

【html】
body内の作成

【css】
2行目
:rootの追加

5行目
フォントサイズのため追加

16行目以降　作成

tommyKazuhito commented on basic-layout/no4/htdocs/index.html:
```diff
@@ -15,7 +17,88 @@
+      <header class="header">
+        <p class="txt">HEADER</p>
+      </header>
+      <div class="main-visual">
+        <p class="txt">MAIN VISUAL</p>
+      </div>
```
@andoazumi 
この要素は `main` の中に入れたほうがいいかなと思います。

flexで高さ埋めのためにこのマークアップをしていると思いますが、
仮に下層ページがあった場合、大枠のレイアウトを変える必要が出てきちゃいますよね？
イメージ付きますか？例えば他ページでメインビジュアルがない。或いは高さが違う場合に
`.top-wrapper` の `100vh` の指定がかなり悪さしちゃいます。

でもテクニックとして、このページ単体で見れば全然ありですよ！

tommyKazuhito reviewed (commented):
@andoazumi 
確認しました！
レイアウトの組み方的には、問題ないかなと思います。

カラムの組み方も個人的にはこのやり方が現状、一番いいのかなと思います。

ちょっとコメントしました！

tommyKazuhito commented:
@andoazumi 

細かいですが、プルリクのタイトル、フォーマットに合わせて編集しておいてくださいm(__)m

andoazumi commented:
@tommyKazuhito 
ご確認いただき、ありがとうございます！
タイトルをフォーマット通りにしてないの確認不足です。申し訳ございません。取り急ぎ変更いたしました。

下層ページにも持ってこれるレイアウトという想定が頭に入っておりませんでした。わかりやすく例えで説明いただき、ありがとうございます。
レイアウトが変わっても大丈夫になることを考えながら、MAIN VISUALのところをmainの中へ修正いたします。

lineatakano commented on basic-layout/no4/htdocs/css/style.css:
```diff
@@ -14,3 +29,262 @@ body {
+
+  .inner .contents-02 {
+    height: 250px;
+    width: calc(50% - 20px / 2);
```
@andoazumi 
確認遅くなりました。

親でgap: 20px;を指定しているので、ここのwidth: calc(50% - 20px / 2)は問題ないですか？
確認をいただければと！

andoazumi replied:
@lineatakano 
ご確認いただき、ありがとうございます！

該当箇所の確認と私がgapとcalcをまだまだ理解していないと思いますので、改めて学習いたします。
ありがとうございます！

andoazumi replied:
@lineatakano 
おつかれさまです。

知識不足でどう書き直したらいいのかわからないので、ヒントをいただけないでしょうか…。
calcの計算は前にも間違えていて、式をどう書いたらいいか考える力が私にはまだまだ足りていないです。
申し訳ございませんが、お手隙の際によろしくお願いいたします。

tommyKazuhito replied:
@andoazumi 
`50% - 20px / 2` の式の数学的？図学的？な意味ってわかります？

lineatakano replied:
@andoazumi 
すみません、返事が遅くて、、、。

gapが指定されているのと、flex-wrapがno-wrapなので結果には影響がでなく
崩れないのでわかりづらいですが、@tommyKazuhito の指摘通りで計算が合っていないようです。

andoazumi replied:
@tommyKazuhito 

意味としては全体の半分から20pxの半分を引いているだと思っているのですが、いかがでしょうか。
二つ並べるために半々で分けつつ、それぞれから余白分の半分を引くことで二つ並んでも100%を超えないようにで考えています。

tommyKazuhito replied:
@andoazumi 
```
意味としては全体の半分から20pxの半分を引いているだと思っているのですが、いかがでしょうか。
二つ並べるために半々で分けつつ、それぞれから余白分の半分を引くことで二つ並んでも100%を超えないようにで考えています。
```
あってます！つまり式は正しいんです！
自信持って大丈夫です！

@lineatakano 
flex-wrapの指定、見間違えてましたかね？w

lineatakano replied:
@tommyKazuhito @andoazumi 
すみません、コメントをする箇所が間違ってました。
262行目の「width: calc(25% - 28px / 4)」の方でした
ただ、ほかが計算あってますので、単純にミスでしたね。。。
混乱させてしまいましたね、ほんとすみません！！！

andoazumi replied:
@lineatakano @tommyKazuhito 
ご確認いただきまして、ありがとうございます。
私の方こそお騒がせしてしまって申し訳ございません。

262行目のところを確認しまして修正を提出いたしますので、よろしくお願いいたします！

andoazumi committed:
[Update]課題其の四 『WEBページの基本的なレイアウト2』再　修正

【html】
24行目
.top-wrapper 削除

27行目
.main-visualをmainタグ内に入れるために移動

【css】
14行目
headerの高さの変数を追加

54行目
.top-wrapperを削除したのでこちらも削除いたしました。

93,162,230行目
ヘッダーの高さを変数にしたことに伴い変更

97,167,234行目
.main-visualがmainタグ内に移動したので削除

100,169,237行目
mainタグ内で.main-visualを最初の画面で余白いっぱいに広げるために追加
.mainでつけて作っていた余白を.contents-01-areaでつけるように変更

262行目
contents2を四つ並べると幅が100%を超えてしまっていたので、修正

andoazumi commented:
@tommyKazuhito 
@lineatakano 

おつかれさまです。
ご指摘いただいた箇所を修正いたしましたので、お手隙の際にご確認の程よろしくお願い申し上げます。

tommyKazuhito commented on basic-layout/no4/htdocs/css/style.css:
```diff
@@ -12,6 +12,9 @@
   --fz-sp: 2.4rem;
   --fz-tab: 2.8rem;
   --fz-pc: 3.6rem;
+  --height-sp: 85px;
+  --height-tub: 100px;
+  --height-pc: 120px;
```
@andoazumi 
ヘッダーの高さのことだと思うのですが、
`--header-**` だとなんの高さのことか分からないので、変数の命名として良くないかなと思います。

例えば、フォントサイズとルールを合わせるなら、
`--h-header-**` や `--header-h-**`
にしてあげたほうがいいと思います。
あくまで例ですが。

lineatakano replied:
@andoazumi 
修正ありがとうございます。
CONTENTS3-3は元々出来ていたようなので、問題なかったですね。

tommyKazuhito reviewed (commented):
@andoazumi 
確認しました！
HTMLの構造はいいと思います！

一箇所CSSでコメントつけましたので、ご確認をお願いいたしますm(__)m

andoazumi commented:
@tommyKazuhito 
ご確認いただき、ありがとうございます！

確かに時間が経ってから見直した時に、何のことだか自分がわからなくなるだろうなって思いました。他のとルールを合わせて覚えるためにも修正します。

andoazumi committed:
[Update]課題其の四 『WEBページの基本的なレイアウト2』再　修正

【css】
15,16,17行目
何の変数なのかわかりやすくするために変数名を変更

90,98,166,174,241,249行目
変数名の変更に伴い修正

andoazumi commented:
@tommyKazuhito 
@lineatakano 

おつかれ様です。
cssの修正をいたしましたので、お手隙の際にご確認の程よろしくお願いいたします。

tommyKazuhito reviewed (approved):
@andoazumi 
LGTM!
